draft-ietf-httpbis-http2-secondary-certs.md

```diff
@@ -753,6 +753,80 @@ any stream which references the failing certificate in question or process the
 requests as unauthenticated and provide error information at the HTTP semantic
 layer.
 
+# Certificate Extensions {#extension}
+
+## Allowed Primary Domain Certificate Extension {#extension-primary}
+
+The Allowed Primary Domain extension allows certificates to limit their use with
+Secondary Certificate Authentication.  The identities listed in this extension
+are restrictions asserted by the requester of the certificate and are not
+verified by the CA.  Conforming CAs SHOULD mark the allowedPrimaryDomain
+extension as non-critical.
+
+Each allowed primary domain is represented as an IA5String. The name MUST be in
+the "preferred name syntax", as specified by Section 3.5 of {{!RFC1034}} and as
+modified by Section 2.1 of {{!RFC1123}}.  Note that while uppercase and
+lowercase letters are allowed in domain names, no significance is attached to
+the case. In addition, while the string " " is a legal domain name,
+affiliatedDomain extensions MUST NOT contain " ".  Rules for encoding
+internationalized domain names are specified in Section 7.2 of {{!RFC5280}}.
+
+If the affiliatedDomain extension is present, the sequence MUST contain at least
+one entry.  Unlike the subject field, conforming CAs MUST NOT issue certificates
+with an affiliatedDomain extension containing empty GeneralName fields.  Clients
+that encounter such a certificate when processing a certification path MUST
+ignore that extension.
+
+The wildcard character "*" MAY be used to represent that any set of subdomains
+is acceptable.  This character MUST be the entirety of the first label if used,
+and MUST NOT occur elsewhere.  (That is, "*.example.com" and "*" are acceptable,
+but "www.*.com" and "w*.example.com" are not).
+
+    id-ce-allowedPrimaryDomain OBJECT IDENTIFIER ::=  { id-ce TBD1 }
+
+    AllowedPrimaryDomain ::= SEQUENCE SIZE (1..MAX) OF IA5String
```

Comment:
Why not structure this like subjectAltName and allow alternative identifier types as well?  Too much feature?  The advantage would be that you could cite the subjectAltName definition whole cloth and concentrate on matching rules.

Reply:
Yeah, mostly too much feature.  The security properties are probably similar -- if you can hijack a domain to fool DV, you can hijack any e-mail address on the domain to fool an e-mail validation, etc.  We expect domains to be the common case and they're sufficient, so let's not multiply entities without need.